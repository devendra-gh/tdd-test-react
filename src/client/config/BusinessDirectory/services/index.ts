import getMapNatureData from './getMapNatureData';
import getMapActivityData from './getMapActivityData';
import getToken from './getToken';

/* istanbul ignore file */

export { getMapActivityData, getMapNatureData, getToken };
