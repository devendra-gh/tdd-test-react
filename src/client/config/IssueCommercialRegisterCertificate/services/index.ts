import { IVariables } from '@tamm/app-composer';
import fetch from 'client/services/fetch';
import baseUrl from 'client/utils/baseUrl';
import { PATH_ERROR } from '../routes';

export const fetchLicences = async () => {
  return fetch('/pub/proxy/listTradeLicensesV3', 'POST', {}).then(
    (response: IVariables) => {
      let licences = [];
      const {
        responseStatus: { statusCode },
      } = response.data;
      if (statusCode === 200) {
        licences = response.data.TradeLicensesList.contents;
      }
      return licences;
    },
  );
};

export const issueCertificate = async ({ cnNumber }: IVariables) => {
  return fetch('/pub/proxy/issueLetter', 'POST', {
    cnNumber,
    letterType: 'license',
  }).then(({ success, data }) => {
    if (success) {
      return data;
    }
    return null;
  });
};

export const downloadBusinessCertificate = async ({
  instanceId,
  certificateName,
  type,
}: IVariables) => {
  window.open(
    `${baseUrl}/api/download/businessCertificateGenericAuthADU?instanceId=${instanceId}&type=${type}&certificateName=${certificateName}&mobileDownloadable=pdf&mobileFileName=Certificate`,
  );
};

export const errorBoundary = (props: IVariables, error: any) => {
  // console.log('error fetching user licenses', error); // eslint-disable-line
  props.history.push(PATH_ERROR);
};
