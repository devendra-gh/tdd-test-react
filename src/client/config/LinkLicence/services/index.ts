import fetch from 'client/services/fetch';

// TODO: get the details about the licence
const industrialLicenceDetails = (licenseNo: string) => {
  return fetch('/pub/proxy/getLicenceDetailsV3', 'POST', {
    licenseNo: licenseNo || '',
  })
    .then(response => {
      const details = {
        code: response.success ? '204' : '500',
        message: response.data.message,
        traceId: response.data.traceId,
      };
      if (response.success && response.data && response.data.result) {
        if (response.data.result.constructor === Array) {
          const data = response.data.result[0];
          Object.assign(details, {
            code: response.data.code,
            licenceNumber: data.licenseNo,
            licenceTypeEn: data.clasification_en,
            licenceTypeAr: data.clasification_ar,
            tradeNameEn: data.businessNameEng,
            tradeNameAr: data.businessNameArb,
          });
        }
        return details;
      }
      throw details;
    })
    .catch(err => {
      throw err;
    });
};

// TODO: get link licence status
const linkUserLicence = async (
  licenseNo: string,
  spuuid: string,
  userEmail: string,
  userMobile: string,
) => {
  try {
    return await fetch('/pub/proxy/linkLicense', 'POST', {
      licenseNumber: licenseNo || '',
      SPUUID: spuuid || '',
      userEmail,
      userMobile,
    });
  } catch (error) {
    throw error;
  }
};

export const services: { [key: string]: Function } = {
  industrialLicenceDetails,
  linkUserLicence,
};
