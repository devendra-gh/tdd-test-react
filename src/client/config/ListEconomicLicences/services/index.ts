import { IVariables } from '@tamm/app-composer';
import fetch from 'client/services/fetch';
import baseUrl from 'client/utils/baseUrl';
import { PATH_ERROR } from '../routes';

export const issueCertificate = async ({ cnNumber }: IVariables) => {
  return fetch('/pub/proxy/issueLetter', 'POST', {
    cnNumber,
    letterType: 'haslicense',
  }).then(({ success, data }) => {
    if (success) {
      return data;
    }
    return null;
  });
};

export const downloadBusinessCertificate = async ({
  instanceId,
  certificateName,
  type,
}: IVariables) => {
  window.open(
    `${baseUrl}/api/download/businessCertificateGenericAuthADU?instanceId=${instanceId}&type=${type}&certificateName=${certificateName}&mobileDownloadable=pdf&mobileFileName=Certificate`,
  );
};

export const errorBoundary = (props: IVariables, error: any) => {
  props.history.push(PATH_ERROR);
};
