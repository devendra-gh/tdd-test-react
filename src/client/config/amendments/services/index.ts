import getLicenceDetails from './getLicenceDetails';
import getActivities from './getActivities';
import getRelatedRecords from './getRelatedRecords';
import checkApplicationStatus from './checkApplicationStatus';
import updateReduxState from './updateReduxState';
import getReduxState from './getReduxState';

export {
  getLicenceDetails,
  getActivities,
  getRelatedRecords,
  checkApplicationStatus,
  updateReduxState,
  getReduxState,
};
