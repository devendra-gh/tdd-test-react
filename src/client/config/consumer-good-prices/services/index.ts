import fetchGoodsFromSearchText from './goodsSearch';
import fetchGoodsDetail from './getItem';
import fetchCategories from './categories';
import fetchGoodsFromCategory from './goodsCategories';

export {
  fetchGoodsFromSearchText,
  fetchGoodsDetail,
  fetchCategories,
  fetchGoodsFromCategory,
};
