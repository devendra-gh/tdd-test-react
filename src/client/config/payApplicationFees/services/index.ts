import checkApplicationStatus from './checkApplicationStatus';
import fetchLicenceDetails from './fetchLicenceDetails';
import applicationFees from './applicationFees';
import getVariables from './bpm';

export default {
  checkApplicationStatus,
  fetchLicenceDetails,
  applicationFees,
  getVariables,
};
