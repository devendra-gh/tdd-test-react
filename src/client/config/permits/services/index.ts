import fetch from 'client/services/fetch';
import { permitType } from '../utils/constants';

export const fetchRequirementsInquiry = async (serviceName: string) => {
  return fetch('/api/proxy/permitRequirementsInquiry', 'POST', {
    permitType: permitType[serviceName].id,
  });
};
