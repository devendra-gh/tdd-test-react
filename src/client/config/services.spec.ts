import services, { emptyFunc, goToPage } from './services';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('config/services', () => {
  const props = {
    history: {
      push: jest.fn(),
    },
  };
  const heroProps = { test: 1 };

  it('should call history and emptyFunc', () => {
    expect(emptyFunc()).toMatchSnapshot();
    expect(
      goToPage({ push: jest.fn(page => page) }, '/login')(),
    ).toMatchSnapshot();
  });

  it('should call function', async () => {
    services.forEach(service => {
      const mockMath = Object.create(global.Math);
      mockMath.random = () => 0.7;
      global.Math = mockMath;
      const user = { name: 'test' };
      expect(service.getProps(props, user, heroProps)).toMatchSnapshot();
      expect(service.getProps(props, false, heroProps)).toMatchSnapshot();

      mockMath.random = () => 0.3;
      global.Math = mockMath;
      expect(service.getProps(props, user, heroProps)).toMatchSnapshot();
      expect(service.getProps(props, false, heroProps)).toMatchSnapshot();
    });
  });
});
