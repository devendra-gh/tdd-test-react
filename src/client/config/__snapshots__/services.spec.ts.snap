// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/services should call function 1`] = `
Object {
  "hero": Object {
    "ar": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
    "en": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
  },
  "startLogin": Object {
    "ar": Object {
      "buttonLabel": "Start AR",
      "description": "Start a new service AR",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Start AR",
    },
    "en": Object {
      "buttonLabel": "Start",
      "description": "Start a new service",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Start",
    },
  },
}
`;

exports[`config/services should call function 2`] = `
Object {
  "hero": Object {
    "ar": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
    "en": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
  },
  "startLogin": Object {
    "ar": Object {
      "buttonLabel": "login AR",
      "description": "Please login AR",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Please login AR",
    },
    "en": Object {
      "buttonLabel": "login",
      "description": "Please login",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Please login",
    },
  },
}
`;

exports[`config/services should call function 3`] = `
Object {
  "hero": Object {
    "ar": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
    "en": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
  },
  "startLogin": Object {
    "ar": Object {
      "buttonLabel": "continue AR",
      "description": "Continue to existing service AR",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Continue AR",
    },
    "en": Object {
      "buttonLabel": "continue",
      "description": "Continue to existing service",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Continue",
    },
  },
}
`;

exports[`config/services should call function 4`] = `
Object {
  "hero": Object {
    "ar": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
    "en": Object {
      "aspectOfLifeType": "",
      "backgroundBase64Extension": "",
      "backgroundImage": "https://journeys-stg.tamm.abudhabi/journeys/journey-template/images/hero.jpg",
      "breadcrumbs": Array [],
      "test": 1,
    },
  },
  "startLogin": Object {
    "ar": Object {
      "buttonLabel": "login AR",
      "description": "Please login AR",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Please login AR",
    },
    "en": Object {
      "buttonLabel": "login",
      "description": "Please login",
      "onClick": [Function],
      "stickToTop": true,
      "title": "Please login",
    },
  },
}
`;

exports[`config/services should call history and emptyFunc 1`] = `[Function]`;

exports[`config/services should call history and emptyFunc 2`] = `"/login"`;
